Source/Core/Common/SettingsHandler.cpp

```diff
@@ -96,22 +95,33 @@ void SettingsHandler::Reset()
   m_buffer = {};
 }
 
-void SettingsHandler::AddSetting(std::string_view key, std::string_view value)
+void SettingsHandler::AddSetting(const std::string& key, const std::string& value)
 {
-  for (const char& c : key)
-  {
-    WriteByte(c);
-  }
+  WriteLine(key + '=' + value + "\r\n");
+}
 
-  WriteByte('=');
+void SettingsHandler::WriteLine(const std::string& str)
+{
+  const u32 old_position = m_position;
+  const u32 old_key = m_key;
 
-  for (const char& c : value)
-  {
+  // Encode and write the line
+  for (char c : str)
     WriteByte(c);
-  }
 
-  WriteByte(13);
-  WriteByte(10);
+  // If the encoded data contains a null byte, Nintendo's decoder will stop at that null byte
+  // instead of decoding all the data. To avoid this: If the data we just wrote contains
+  // a null byte, add an LF right before the line to prod the values into being different,
+  // just like Nintendo does. Due to the chosen key, LF itself never encodes into a null byte.
+  const auto begin = m_buffer.cbegin() + old_position;
+  const auto end = m_buffer.cbegin() + m_position;
+  if (std::find(begin, end, 0) != end)
+  {
+    m_key = old_key;
+    m_position = old_position;
+    WriteByte('\n');
+    WriteLine(str);
```

Comment:
Hmm, is it possible for a string to always contain a null byte after encoding, no matter how many line feeds are added to change the key?

Reply:
I don't think that's likely, but I haven't proven it to be impossible (unlike LF never turning into a null byte, which I did prove with a small program).

Reply:
I think that theoretically that could be possible, but I think it's highly unlikely. But in case it does happen, it might be a good idea to add some kind of fallback so that it only tries 32 times (all possible key bits, adding up to 31 newlines) and then prints an error to the log instead of trying again in an endless loop like it would now. 

Also, no idea if this is already implemented, if we assume that such a thing can happen it might also be a good idea to make sure that the encoded file doesn't exceed the maximum file size.

Reply:
> Also, no idea if this is already implemented, if we assume that such a thing can happen it might also be a good idea to make sure that the encoded file doesn't exceed the maximum file size.

This is implemented. `WriteByte` does nothing if the file is full. This also means that we never will get an infinite loop, because if we recurse many times, `begin` will eventually be equal to `end`, making it impossible to recurse further.

Reply:
Oh, right, missed that part.